### PR TITLE
Apply does not release user-defined memory

### PR DESCRIPTION
**Pull request by QingleiCao**
https://github.com/ICLDisco/parsec/issues/675

### CODE REVIEW

**Review by devreal — APPROVED** [review]
Sounds reasonable not to transfer ownership to `parsec_apply`

**Review by bosilca — APPROVED** [review]

**Review by abouteiller — APPROVED** [review]

File: tests/apps/stencil/testing_stencil_1D.c [hunk 1]
```diff
@@ -162,6 +162,7 @@ int main(int argc, char *argv[])
     parsec_apply( parsec, PARSEC_MATRIX_FULL,
                   (parsec_tiled_matrix_t *)&dcA,
                   (parsec_tiled_matrix_unary_op_t)stencil_1D_init_ops, op_args);
```
Comment (abouteiller) [inline tests/apps/stencil/testing_stencil_1D.c:164]:
```suggestion
                  (parsec_tiled_matrix_unary_op_t)stencil_1D_init_ops, &R);
```

File: tests/collections/redistribute/testing_redistribute_random.c [hunk 2]
```diff
@@ -212,11 +210,10 @@ int main(int argc, char *argv[])
         /* 
          * Init dcT to 0.0 for DTD 
          */
-        int *op_args_dtd = (int *)malloc(sizeof(int));
-        *op_args_dtd = 0;
+        int *op_args_dtd = 0;
```
Comment (devreal) [inline tests/collections/redistribute/testing_redistribute_random.c:213]:
```suggestion
        int op_args_dtd = 0; 
```

**Review by devreal — CHANGES_REQUESTED** [review]

File: tests/collections/redistribute/testing_redistribute_random.c (hunk 2 above)
Comment (QingleiCao) [reply tests/collections/redistribute/testing_redistribute_random.c]: fixed. thanks

**Review by devreal — APPROVED** [review]

**Review by therault — APPROVED** [review]

**Pull request merged**